File: testcgi.py
#!/usr/bin/env python3

import cgi

print("Content-Type: text/html")
print()  # 空行でHTTPヘッダーとボディを区切る

print("<html>")
print("<head><title>Hello CGI</title></head>")
print("<body>")
print("<h1>Hello from CGI!</h1>")
print("</body>")
print("</html>")

File: testcgi.cpp
#include <iostream>
#include <cstdlib>

int main()
{
    // CGIのレスポンスヘッダ
    std::cout << "Content-Type: text/html\n\n";

    // HTMLの出力
    std::cout << "<html><head><title>CGI Response</title></head>\n";
    std::cout << "<body>\n";
    std::cout << "<h1>CGI Program Executed</h1>\n";
    std::cout << "<p>You have clicked the button.</p>\n";
    std::cout << "</body></html>\n";

    return 0;
}
